### PR TITLE
Harden payment notification reception path

### PR DESCRIPTION
**Pull request by Lymkwi**
Remove a lot of raw accesses to the data dictionary in the payment notification view, and check them for presence and type when used, with logging added.

---

You've really gotta help me review this one. The code here has to work exactly the same as it did before, but without the vulnerable dictionary queries. One of the step of hardening was determining types, and without being able to deploy and test, I had to rely on past behaviour + documentation.

**We really have to make sure this code works as it used to**

### CODE REVIEW

**Comment by KwikKill** [comment]
I'll deploy it on the beta server tomorrow or later to test it with the sandbox.

File: insalan/payment/views.py [hunk 1]
```diff
@@ -71,30 +71,62 @@ def post(self, request):
...
+        data = data.get("data", default={})
 
         if ntype == "Order":
             # From "Order", get the payments
-            order_id = data["id"]
+            # Check that the id is an integer
+            order_id = data.get("id")
+            if not isinstance(order_id, int):
+                logger.error(
+                    "Invalid ID for Order event, except integer, got %s", order_id
```
Comment (Lugrim) [inline insalan/payment/views.py:91]: Can be a bit ambiguous if it is a string containing only digits as it won't print quotes

**Review by Lugrim — APPROVED** [review]
LGTM, but haven't tested the code, only read it

**Review by KwikKill — APPROVED** [review]
I deployed it in beta and I was about to test it but helloasso broke the sandbox.
But the code seems alright to me, should not break anything.

**Comment by Lymkwi** [comment]
> [...] the code seems alright to me, should not break anything.

Famous last words but I guess only time will tell :laughing:

**Pull request merged**